v3/generate.go

```diff
@@ -62,6 +69,37 @@ func Generate() {
 	}
 }
 
+// duplicateMap creates a deep copy of the provided map
+func duplicateMap(m map[interface{}]interface{}) map[interface{}]interface{} {
```

Comment:
Looks fine, but wondering if there's already an existing utility method out there

Reply:
I can't seem to find any deep copy functions in public libraries. 
The standard library has [map.Copy](https://pkg.go.dev/maps@master#Copy) and [map.Clone](https://pkg.go.dev/maps@master#Clone) but these perform assignment copies or shallow copies. Have you come across one?

Reply:
Hmm, maybe https://stackoverflow.com/a/67139854 but comments say that performance is way worse. Having your own deep copy method is fine. I was just surprised that we haven't done this before